foyer/general_forcefield.py

```diff
@@ -0,0 +1,539 @@
+            Options to speed up if there are a lot of repeated
+            subtopology within the topology (assuming they all
+            have the same name).
+        assert_bond_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system bonds.
+        assert_angle_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system angles.
+        assert_dihedral_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system dihedrals.
+        assert_improper_params : bool, optional, default=True
+            If True, Foyer will exit if parametes are not found for all
+            system impropers.
+        combining_rule : str, optional, default='geometric'
+            The combining rule of the system, stored as an attribute of the
+            GMSO topology.
+        verbose : bool, optional, default=False
+            If True, Foyer will print debug-level information about notable or
+            potential problematic details it encounters.
+        backend : str, optional, default='gmso'
+            Name of the backend used to store all the Types' information.
+            At this point, 'gmso' is the only valid backend, but this set up
+            allow future backend to be implemented easier.
+        """
+        if self.atomTypeDefinitions == {}:
+            raise FoyerError('Attempting to atom-type using a forcefield '
+                'with no atom type definitions.')
+
+        if backend == 'gmso':
+            if not isinstance(top, gmso.Topology):
+                mb = import_('mbuild')
+                tmp_top = mb.load(top)
+                top = gmso.external.from_mbuild(top)
+
+            assert isinstance(top, gmso.Topology)
+            typemap = self._run_atomtyping(top,
+                    use_residue_map=use_residue_map,
+                    **kwargs)
+
+        return self._parametrize(top=top, typemap=typemap,
+            references_file=references_file,
+            assert_bond_params=assert_bond_params,
+            assert_angle_params=assert_angle_params,
+            assert_dihedral_params=assert_dihedral_params,
+            assert_improper_params=assert_improper_params,
+            verbose=verbose,
+            backend=backend,
+            combining_rule=combining_rule,
+            *args, **kwargs)
+
+    def _run_atomtyping(self, top, use_residue_map=True, **kwargs):
+        """Atomtype the topology
+
+        Parameters
+        ----------
+        top : gmso.Topology
+            Molecular Topology to find atom types for.
+        use_residue_map : bool, optional, default=True
+            Spped up options for duplicates subtopology.
+        """
+
+        if isinstance(top, mb.Compound):
+            top = from_mbuild(top)
+        # TO DO in another PR
+        if use_residue_map:
```

Comment:
Maybe I just don't understand what is going on here.  However, it seems that it is doing the same function regardless of whether use_residue_map is True or False.

Reply:
This is a feature that I need to implement in the future (when we flesh out the `Subtopology` class designs or come up with a better tags system). This is just a place holder right now, so having it True or False does not make any different.